src/lexer/Tokenizer.ts

```diff
@@ -105,21 +95,57 @@ export default class Tokenizer {
         ]),
       },
     });
+  }
 
-    this.engine = new TokenizerEngine(rules);
+  // These rules can't be blindly cached as the paramTypesOverrides object
+  // can differ on each invocation of the format() function.
+  private buildParamRules(
+    cfg: TokenizerOptions,
+    paramTypesOverrides: ParamTypes
+  ): Partial<Record<TokenType, TokenRule>> {
+    // Each dialect has its own default parameter types (if any),
+    // but these can be overriden by the user of the library.
+    const paramTypes = {
```

Comment:
would it be better to merge the custom additions instead of just replacing the default behaviour ?

Reply:
I intentionally opted to not merge them to offer more flexibility:

- With overriding one can also turn off the built-in param types, in case they conflict with something. Not sure if that's needed, but with merging it wouldn't be possible.
- On the other hand the merging itself might not really be useful, as one would likely anyway use just one syntax for parameters, not mix several.
- The code for merging would also be more complex.